[api-minor] Implement basic support for OptionalContent `Usage` dicts (issue 5764, bug 1826783)

Snuffleupagus opened the pull request:
The following are some highlights of this patch:
 - In the Worker we only extract a *subset* of the potential contents of the `Usage` dictionary, to avoid having to implement/test a bunch of code that'd be completely unused in the viewer.

 - In order to still allow the user to *manually* override the default visible layers in the viewer, the viewable/printable state is purposely *not* enforced during initialization in the `OptionalContentConfig` constructor.

 - Printing will now always use the *default* visible layers, rather than using the same state as the viewer (as was the case previously).
   This ensures that the printing-output will correctly take the `Usage` dictionary into account, and in practice toggling of visible layers rarely seem to be necessary except in the viewer itself (if at all).[1]

---
[1] In the unlikely case that it'd ever be deemed necessary to support fine-grained control of optional content visibility during printing, some new (additional) UI would likely be needed to support that case.

Snuffleupagus committed:
[api-minor] Implement basic support for OptionalContent `Usage` dicts (issue 5764, bug 1826783)

The following are some highlights of this patch:
 - In the Worker we only extract a *subset* of the potential contents of the `Usage` dictionary, to avoid having to implement/test a bunch of code that'd be completely unused in the viewer.

 - In order to still allow the user to *manually* override the default visible layers in the viewer, the viewable/printable state is purposely *not* enforced during initialization in the `OptionalContentConfig` constructor.

 - Printing will now always use the *default* visible layers, rather than using the same state as the viewer (as was the case previously).
   This ensures that the printing-output will correctly take the `Usage` dictionary into account, and in practice toggling of visible layers rarely seem to be necessary except in the viewer itself (if at all).[1]

---
[1] In the unlikely case that it'd ever be deemed necessary to support fine-grained control of optional content visibility during printing, some new (additional) UI would likely be needed to support that case.

Snuffleupagus commented:
/botio-linux preview

calixteman commented on src/core/catalog.js:
```diff
@@ -469,7 +459,64 @@ class Catalog {
+    }
+
+    const usage = group.get("Usage");
+    if (usage instanceof Dict) {
```
nit: use the negation and a return in order to reduce the indentation.

Snuffleupagus committed:
Move the /SetOCGState handling into the `OptionalContentConfig` class (PR 15377 follow-up)

This helps ensure that /SetOCGState actions always take the `Usage` dictionary into account as expected.

calixteman reviewed (approved):
Overall it looks good to me. Thank you.

Snuffleupagus commented:
/botio test

Snuffleupagus commented:
/botio makeref
